### PR TITLE
fix(mongodb): add missing query logs

### PR DESCRIPTION
**Pull request by laplab**

### CODE REVIEW

File: query-engine/connectors/mongodb-query-connector/src/root_queries/write.rs [hunk 1]
```diff
@@ -495,12 +497,22 @@ pub async fn query_raw<'conn>(
...
+                        let query_string_builder = Find::new(&unwrapped_filter, &project, coll.name());
+                        let cursor = observing(&query_string_builder, || {
+                            coll.find_with_session(filter, options, session)
+                        })
+                        .await?;
 
                         raw::cursor_to_json(cursor, session).await?
                     }
                     MongoOperation::Aggregate(pipeline, options) => {
-                        let cursor = coll.aggregate_with_session(pipeline, options, session).await?;
```
Comment (laplab) [inline query-engine/connectors/mongodb-query-connector/src/root_queries/write.rs:503]: I `grep`ed the codebase and these two places seem to be the only ones missing an `observing` call. Let me know if you find anything else

Comment (Weakky) [reply query-engine/connectors/mongodb-query-connector/src/root_queries/write.rs:503]: Couldn't find any more either 👍

File: query-engine/connectors/mongodb-query-connector/src/root_queries/mod.rs [hunk 2]
```diff
@@ -72,13 +66,9 @@ where
...
     increment_counter!(PRISMA_DATASOURCE_QUERIES_TOTAL);
 
-    let params: Vec<i32> = Vec::new();
-
-    // todo: emit tracing event with the appropriate log level only query_log is enabled. And fix log suscription
-    if let Some(qs) = query_string_builder {
-        let qs = qs.build();
-        debug!(target: "mongodb_query_connector::query", item_type = "query", is_query = true, query = %qs, params = ?params, duration_ms=elapsed);
-    }
+    // TODO: emit the event only when the query logs are enabled.
```
Comment (Weakky) [inline query-engine/connectors/mongodb-query-connector/src/root_queries/mod.rs:69]: That TODO rewording removes a lot of the info that was initially put. Any reason why you removed a lot of it?

File: query-engine/connectors/mongodb-query-connector/src/root_queries/write.rs [hunk 3]
```diff
@@ -495,12 +497,22 @@ pub async fn query_raw<'conn>(
 
                 match operation {
                     MongoOperation::Find(filter, options) => {
-                        let cursor = coll.find_with_session(filter, options, session).await?;
+                        let unwrapped_filter = filter.clone().unwrap_or_default();
+                        let project = Document::default();
```
Comment (Weakky) [inline query-engine/connectors/mongodb-query-connector/src/root_queries/write.rs:501]: Why are we always passing an empty projection instead of using the projection in the `FindOptions`?

File: query-engine/connectors/mongodb-query-connector/src/root_queries/mod.rs [hunk 4]
```diff
@@ -47,20 +47,14 @@ fn pick_singular_id(model: &Model) -> ScalarFieldRef {
...
-    f: F,
-) -> mongodb::error::Result<T>
+/// Logs the query and updates metrics for an operation performed by a passed function.
+///
+/// NOTE:
+/// 1. `dyn QueryString` is used instead of a `String` to skip expensive query serialization when
+///    query logs are disabled. This, however, is not currently implemented.
+/// 2. Query strings logged are for debugging purposes only. The actual queries constructed by
+///    MongoDB driver might look slightly different.
+pub(crate) async fn observing<'a, 'b, F, T, U>(builder: &'b dyn QueryString, f: F) -> mongodb::error::Result<T>
```
Comment (Weakky) [inline query-engine/connectors/mongodb-query-connector/src/root_queries/mod.rs:57]: Nice that we made the builder required 👍

File: query-engine/connectors/mongodb-query-connector/Cargo.toml [hunk 5]
```diff
@@ -21,8 +21,9 @@ tracing = "0.1"
 tracing-futures = "0.2"
 uuid.workspace = true
 indexmap = "1.7"
-query-engine-metrics = {path = "../../metrics"}
+query-engine-metrics = { path = "../../metrics" }
 cuid = { git = "https://github.com/prisma/cuid-rust", branch = "wasm32-support" }
+derive_more = "0.99.17"
```
Comment (Weakky) [inline query-engine/connectors/mongodb-query-connector/Cargo.toml:26]: I'm a bit meh on the idea of bringing a new crate for the single purpose of avoiding a couple of `Self::new` methods. I see it has a bunch of other macros that could be useful. Let's try it.

**Review by Weakky — APPROVED** [review]
A couple of remarks, looks good 👍

File: query-engine/connectors/mongodb-query-connector/Cargo.toml (hunk 5 above)
Comment (laplab) [reply query-engine/connectors/mongodb-query-connector/Cargo.toml:26]: I agree. We have used this crate a lot for other purposes in my previous project, so I also hope that it will come more useful than just for this.

File: query-engine/connectors/mongodb-query-connector/src/root_queries/mod.rs (hunk 2 above)
Comment (laplab) [reply query-engine/connectors/mongodb-query-connector/src/root_queries/mod.rs:69]: I was meaning to ask you about it, but totally forgot. Do you know what the author meant by "fix log subscription"?

I propose to reword it to:
```
// TODO: emit the event only when "debug" level query logs are enabled.
```

Let me know if you have a better version in mind.
